src/components/Journal/JournalEntryForm/JournalEntryForm.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { forwardRef, useCallback, useImperativeHandle, useEffect } from 'react'
+import { useJournalEntryForm } from './useJournalEntryForm'
+import { useJournalNavigation } from '../../../providers/JournalStore/JournalStoreProvider'
+import { Form } from '../../ui/Form/Form'
+import { HStack, VStack } from '../../ui/Stack/Stack'
+import { JournalEntryLineItemsTable } from './JournalEntryLineItemsTable'
+import { DataState, DataStateStatus } from '../../DataState'
+import { AlertTriangle } from 'lucide-react'
+import { TextSize } from '../../Typography'
+import { flattenValidationErrors } from '../../../utils/form'
+import { LedgerEntryDirection } from '../../../schemas/generalLedger/ledgerAccount'
+import { UpsertJournalEntryMode } from './useUpsertJournalEntry'
+import './journalEntryForm.scss'
+import { usePreloadCustomers } from '../../../features/customers/api/useListCustomers'
+import { usePreloadVendors } from '../../../features/vendors/api/useListVendors'
+import { CustomerVendorSelector } from '../../../features/customerVendor/components/CustomerVendorSelector'
+import { TagDimensionsGroup } from './TagDimensionsGroup'
+import type { CustomerVendorSchema } from '../../../features/customerVendor/customerVendorSchemas'
+import { Separator } from '../../Separator/Separator'
+
+type CustomerVendor = typeof CustomerVendorSchema.Type
+
+const JOURNAL_ENTRY_FORM_CSS_PREFIX = 'Layer__JournalEntryForm'
+
+export type JournalEntryFormState = {
+  isDirty: boolean
+  isSubmitting: boolean
+}
+
+export type JournalEntryFormProps = {
+  isReadOnly?: boolean
+  onSuccess?: () => void
+  onChangeFormState?: (formState: JournalEntryFormState) => void
+  showTags?: boolean
+  showCustomerVendor?: boolean
+}
+
+export const JournalEntryForm = forwardRef<{ submit: () => Promise<void> }, JournalEntryFormProps>((props, ref) => {
+  const { toJournalTable } = useJournalNavigation()
+
+  const { isReadOnly = false, onSuccess, onChangeFormState, showTags = true, showCustomerVendor = true } = props
+
+  usePreloadCustomers()
+  usePreloadVendors()
```

Comment:
I think we can pass this, and should also preload tags.
```suggestion
  usePreloadCustomers({ isEnabled: showCustomerVendor })
  usePreloadVendors({ isEnabled: showCustomerVendor })
```